jax/_src/random.py

```diff
@@ -232,23 +232,24 @@ def fold_in(key: KeyArray, data: IntegerArray) -> KeyArray:
   key, wrapped = _check_prng_key(key)
   return _return_prng_keys(wrapped, _fold_in(key, data))
 
-def _split(key: KeyArray, num: int = 2) -> KeyArray:
+def _split(key: KeyArray, num: Union[int, tuple[int, ...]] = 2) -> KeyArray:
   # Alternative to split() to use within random samplers.
   # TODO(frostig): remove and use split(); we no longer need to wait
   # to always enable_custom_prng
   assert isinstance(key, prng.PRNGKeyArray)
   if key.ndim:
     raise TypeError("split accepts a single key, but was given a key array of"
                     f"shape {key.shape} != (). Use jax.vmap for batching.")
-  return prng.random_split(key, count=num)
+  shape = tuple(num) if isinstance(num, Sequence) else (num,)
+  return prng.random_split(key, shape=shape)
 
-def split(key: KeyArray, num: int = 2) -> KeyArray:
+def split(key: KeyArray, num: Union[int, tuple[int, ...]] = 2) -> KeyArray:
```

Comment:
`core.Shape` is more permissive than `tuple[int, ...]`. It's seems strange that we use the permissive annotation intenally and a stricter annotation on the public API. Should we use the same annotation everywhere?

Reply:
Do we want to allow `core.Shape` ultimately, and so it's what the internals ought to handle?

Reply:
The main difference is that `core.Shape` allows for extensible dimension sizes, e.g. dynamic shapes. Not sure if that's relevant here yet?

Reply:
I've changed the annotations to concrete shapes.